dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemoryAllocation.java

```diff
@@ -89,11 +89,7 @@ private void createInitEvents(Program program) {
         int nextThreadId = Stream.concat(program.getThreads().stream(), program.getFunctions().stream())
                 .mapToInt(Function::getId).max().getAsInt() + 1;
         for(MemoryObject memObj : program.getMemory().getObjects()) {
-            // The last case "heuristically checks" if Smack generated initialization or not:
-            // if any field is statically initialized, then likely everything is initialized.
-            final boolean isStaticallyInitialized = !isLitmus
-                    && memObj.isStaticallyAllocated();
-            final Iterable<Integer> fieldsToInit = isStaticallyInitialized ?
+            final Iterable<Integer> fieldsToInit = !isLitmus ?
```

Comment:
Actually, the old code was already correct and did not have the heuristics anymore (the comment is just outdated).
I didn't notice that before.